src/Application/ConfigResolver.php

```diff
@@ -154,11 +156,23 @@ private static function mergeInputRequirements(array $config, InputInterface $in
         return $config;
     }
 
-    private static function getComposer(string $directory): Composer
+    /**
+     * @param InputInterface $input
+     * @param array<string> $directories
+     *
+     * @return Composer
+     */
+    private static function getComposer(InputInterface $input, array $directories): Composer
     {
-        $composerPath = $directory . DIRECTORY_SEPARATOR . 'composer.json';
+        /** @var string|null $composerPath */
+        $composerPath = $input->hasOption('composer') ? $input->getOption('composer') : null;
+
+        if ($composerPath === null) {
+            $directory = rtrim($directories[0], '/');
```

Comment:
As we use only $directories[0], maybe we should pass it directly to the function ?

Reply:
Done